proxymode/resolver.go

```diff
@@ -16,25 +17,57 @@ type proxyResolver struct {
 	destinationURL url.URL
 }
 
+var acceptedHeaders = map[string]string{
+	"Content-Length": "",
+	"Content-Type":   "",
+	"Last-Modified":  "",
+}
+
+func (pr proxyResolver) pipeRequestProxy(w http.ResponseWriter, remotePath string) error {
+	remoteURL := pr.destinationURL.ResolveReference(&url.URL{Path: remotePath})
+
+	resp, err := http.Get(remoteURL.String())
+	if err != nil {
+		return err
+	}
+	defer resp.Body.Close()
+
+	for header, values := range resp.Header {
+		if len(w.Header().Values(header)) > 0 {
+			continue
+		}
+		if _, ok := acceptedHeaders[header]; !ok {
+			continue
+		}
+		for _, value := range values {
+			w.Header().Add(header, value)
+		}
+	}
+	w.WriteHeader(resp.StatusCode)
+
+	_, err = io.Copy(w, resp.Body)
+	if err != nil {
+		return err
```

Comment:
This should probably produce a 500 error, maybe unless the error is caused because the client has closed the connection.

Reply:
Added a response with `http.StatusInternalServerError` status in case of error.